reactor-core/src/test/java/reactor/core/publisher/FluxUsingTest.java

```diff
@@ -146,6 +146,24 @@ public void normalEager() {
 		assertThat(cleanup).hasValue(1);
 	}
 
+	@Test
+	public void normalAutoCloseable() {
+		AssertSubscriber<AutoCloseable> ts = AssertSubscriber.create();
+
+		AtomicInteger cleanup = new AtomicInteger();
+
+		AutoCloseable resource = cleanup::incrementAndGet;
+
+		Flux.using(() -> resource, r -> Flux.just(resource))
```

Comment:
As a matter of fact, this test can also check `doOnTerminate` condition just like `Mono` does. As a side quest, the existing `normal` and `normalEager` tests in `FluxUsingTest` could leverage this distinction as currently they are the same test basically.

Reply:
For this you can consider parameterized methods not to duplicate the codebase in e.g. `sourceFactoryAndResourceCleanupThrow`

Reply:
✅ updated, I use `ParameterizedTest` to de-dup test codes and add&enhance test cases.

I'm not sure it's what you want, so feel free to share your opnion~!